sdks/python/apache_beam/tests/pipeline_verifiers.py

```diff
@@ -48,3 +55,63 @@ def describe_mismatch(self, pipeline_result, mismatch_description):
     mismatch_description \
       .append_text("Test pipeline job terminated in state: ") \
       .append_text(pipeline_result.current_state())
+
+
+def retry_on_fileio_error(exception):
+  """Filter allowing retries on file I/O errors."""
+  if isinstance(exception, RuntimeError) or \
+          isinstance(exception, IOError):
+    # GCS I/O raises RuntimeError and local filesystem I/O
```

Comment:
Please file a JIRA about this. I am hoping that `ChannelFactory` will return consistent errors for both. And please remove the references to GCS.